### DISCUSSION TITLE
Decrypt RTSPS (RTSP over TLS)

### DISCUSSION THREAD

**Discussion by ghost** [bluenviron/mediamtx#2786] [Questions]
### Question

How to decrypt RTSPS (RTSP over TLS) to obtain an RTSP that can be read by VLC, for example ? 
Has anyone done this with mediamtx or nginx?

**Comment by aler9** [comment]
Hello, the README describes how to convert TLS-based protocols into plain protocols, although the example is for RTMPS it applies to RTSPS too:

> Be aware that RTMPS is currently unsupported by all major players. However, you can use a proxy like [stunnel](https://www.stunnel.org/) or [nginx](https://nginx.org/) or a dedicated MediaMTX instance to decrypt streams before reading them.

Short answer: launch a local instance of MediaMTX on the same machine of VLC and let it perform the conversion from RTSPS to RTSP, in this way:

```yml
paths:
  my_feed:
    source: rtsps://original-mediamtx-url
```

Then read from

```
rtsp://localhost:8554/my_feed
```

Instead of

```
rtsps://original-mediamtx-url
```

**Reply by ghost** [reply]
Hi!

First, Thanks for your answer.

This is what I tried to do (this is working with a classic RTSP)...

```
My RTSPS is : rtsps://username:password@A.B.C.D:5541/tec/SSB_cam1_h264 
```

When I start the server, here are the logs : 
_2023/12/07 10:25:40 INF MediaMTX v1.3.1
2023/12/07 10:25:40 INF configuration loaded from /home/trudemo/Downloads/rtsp-server/mediamtx.yml
2023/12/07 10:25:40 INF [path my_feed] [RTSP source] started
2023/12/07 10:25:40 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/12/07 10:25:40 INF [RTMP] listener opened on :1935
2023/12/07 10:25:40 INF [HLS] listener opened on :8888
2023/12/07 10:25:40 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
2023/12/07 10:25:40 INF [SRT] listener opened on :8890 (UDP)
2023/12/07 10:25:40 ERR [path my_feed] [RTSP source] tls: failed to verify certificate: x509: cannot validate certificate for 192.168.15.162 because it doesn't contain any IP SANs
2023/12/07 10:25:45 ERR [path my_feed] [RTSP source] tls: failed to verify certificate: x509: cannot validate certificate for 192.168.15.162 because it doesn't contain any IP SANs
2023/12/07 10:25:50 ERR [path my_feed] [RTSP source] tls: failed to verify certificate: x509: cannot validate certificate for 192.168.15.162 because it doesn't contain any IP SANs
2023/12/07 10:25:55 ERR [path my_feed] [RTSP source] tls: failed to verify certificate: x509: cannot validate certificate for 192.168.15.162 because it doesn't contain any IP SANs_

Could this be linked to the fact that the RTSPS stream port is 5541? 

Thank you!

**Reply by ghost** [reply]
I had a certificate problem
I fixed it
I appreciate your help !

**Reply by ghost** [reply]
Is there a way to reduce the delay between input and output stream ?

**Reply by aler9** [reply]
Regarding latency in RTSP, take a look here: https://github.com/bluenviron/mediamtx#latency

[a comment by an automated account is not shown]
